R/boarding.R

```diff
@@ -1,28 +1,138 @@
 
 
-#' Contingency table between ward category and main specialty
+#' Contingency table between ward category and main specialty/disease categories.
 #'
-#' @param spell_table
+#' @description
+#' \code{get_ward_specialty_contingency_table} returns a table showing the relationship
+#' between ward types and main specialties or disease groups. Used in patient boarding analysis to infer
+#' what type of patients a given ward is designed to accommodate.
 #'
-#' @return
+#' @param ipData
+#' \itemize{Hospital inpatient episode data with at least the following fields:
+#'   \item \code{ward_category} - category of the ward;
+#'   \item \code{main_specialty} - main specialty of the clinician responsible for the episode;
+#'   \item \code{ccsr_category_description} - disease group that ICD-10 code of the patient belongs to.
+#' }
+#' @param vars Variables of interest - either \code{"ws"} (ward type - main specialty) or
+#' \code{"wd"} (ward type - CCSR disease category).
+#' @param scaleBy
+#' \itemize{Whether and how entries of contingency table should be scaled.
+#'   \item \code{"none"} - no scaling, presents absolute episode counts;
+#'   \item \code{"rowsum"} - normalizes by the total number of episodes in a given ward. Useful when investigating the distribution of specialties/diseases among wards.
+#'   \item \code{"colsum"} - normalizes by the total number of episodes in a given specialty/disease group. Useful when investigating the distribution of wards among specialties/diseases.
+#'   \item \code{"total"} - normalizes by the total number of episodes.
+#' }
+#' @param returnPlot A boolean value indicating whether a plot (heatmap) should be return. If \code{FALSE}, a table is returned.
+#'
+#' @return A contingency table (default) or a heatmap showing the relationship between ward type and
+#' main specialty or CCSR categories.
 #' @export
 #'
 #' @examples
+#' \dontrun{
+#' get_contingency_table(
+#'   ip_data = inpatient_data,
+#'   vars = "ws",
+#'   scaleBy = "none",
+#'   returnPlot = FALSE
+#' )
+#' }
 #'
-get_ward_specialty_contingency_table <- function(ip_data){
-  # Perform check if the necessary column exist
-  if (!("ward_category" %in% colnames(ip_data))) {
-    stop("The 'ward_category' column not found in the provided dataframe.")
+#' @export
+get_contingency_table <- function(ipData, vars, scaleBy = "none", returnPlot = FALSE) {
```

Comment:
Call a fn to join the requisite additional columns here to avoid additional data files